flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/plan/stream/sql/UnionTest.scala

```diff
@@ -112,9 +112,6 @@ class UnionTest extends TableTestBase {
 
   @Test
   def testUnionDiffRowTime(): Unit = {
-    expectedException.expectMessage(
```

Comment:
UnionTest#testUnionDiffRowTime could work fine now.
In DEFAULT_REWRITE phase, after apply CoerceInputsRule(classOf[LogicalUnion], optimizer would pre-casts inputs to a particular type to ensure union set operator have the same row type.